Okta.Xamarin/Okta.Xamarin/OktaContext.cs

```diff
@@ -724,13 +810,21 @@ public virtual async Task<RenewResponse> RenewAsync(bool refreshIdToken = false,
         /// <param name="refreshIdToken">A value indicating whether to renew the ID token, the default is false.</param>
         /// <param name="authorizationServerId">The authorization server id.</param>
         /// <returns>A <see cref="Task{RenewResponse}"/> representing the result of the asynchronous operation.</returns>
-        public virtual async Task<RenewResponse> RenewAsync(string refreshToken, bool refreshIdToken = false, string authorizationServerId = "default")
+        public virtual async Task<RenewResponse> RenewAsync(string refreshToken, bool refreshIdToken = false, string authorizationServerId = null)
         {
-            this.RenewStarted?.Invoke(this, new RenewEventArgs { StateManager = this.StateManager, RefreshToken = refreshToken, RefreshIdToken = refreshIdToken, AuthorizationServerId = authorizationServerId });
-            RenewResponse result = await this.StateManager.RenewAsync(refreshIdToken, authorizationServerId);
-            this.RenewCompleted?.Invoke(this, new RenewEventArgs { StateManager = this.StateManager, RefreshToken = refreshToken, Response = result, RefreshIdToken = refreshIdToken, AuthorizationServerId = authorizationServerId });
+            try
+            {
+                this.RenewStarted?.Invoke(this, new RenewEventArgs { StateManager = this.StateManager, RefreshToken = refreshToken, RefreshIdToken = refreshIdToken, AuthorizationServerId = authorizationServerId });
+                RenewResponse result = await this.StateManager.RenewAsync(refreshIdToken, authorizationServerId);
+                this.RenewCompleted?.Invoke(this, new RenewEventArgs { StateManager = this.StateManager, RefreshToken = refreshToken, Response = result, RefreshIdToken = refreshIdToken, AuthorizationServerId = authorizationServerId });
 
-            return result;
+                return result;
+            }
+            catch (Exception ex)
+            {
+                this.RenewException?.Invoke(this, new RenewExceptionEventArgs { StateManager = this.StateManager, Exception = ex });
+                return null;
```

Comment:
If no one subscribes to the RenewException event, should we still throw the exception? Otherwise, the exception is lost.

Reply:
Throwing in this model could cause the app to crash unexpectedly and in ways nearly impossible to debug.

Reply:
To mitigate risk of losing visibility to exceptions, a logger should/could be subscribed to exception events.  This is something I'll add to documentation.  A backlog item should exist/be created (I'll create if necessary) to provide a default logger implementation.